src/ringmesh/io/geomodel_adapter_resqml.cpp

```diff
@@ -0,0 +1,676 @@
+                    properties[i] = nullptr;
+                    continue;
+                }
+
+                GEO::AttributeStore* store =
+                    attri_manager.find_attribute_store( attribute_names[i] );
+                std::string element_type = GEO::AttributeStore::
+                    element_type_name_by_element_typeid_name(
+                        store->element_typeid_name() );
+                if( !GEO::AttributeStore::element_typeid_name_is_known(
+                        element_type ) )
+                {
+                    Logger::warn( "I/O",
+                        "Skipping attribute: ", attribute_names[i],
+                        " of type: ", element_type );
+                    properties[i] = nullptr;
+                    continue;
+                }
+
+                std::string guid( GuidTools::generateUidAsString() );
+                if( element_type == "double" )
+                {
+                    properties[i] = pck_->createContinuousProperty( &rep, guid,
+                        attribute_names[i], store->dimension(), attachment,
+                        "unitless",
+                        gsoap_resqml2_0_1::
+                            resqml2__ResqmlPropertyKind__continuous );
+                }
+                else if( element_type == "int" )
+                {
+                    properties[i] = pck_->createDiscreteProperty( &rep, guid,
+                        attribute_names[i], store->dimension(), attachment,
+                        gsoap_resqml2_0_1::
+                            resqml2__ResqmlPropertyKind__discrete );
+                }
+            }
+        }
+
+        for( auto i : range( attribute_names.size() ) )
+        {
+            if( properties[i] == nullptr )
+            {
+                continue;
+            }
+            GEO::AttributeStore* store =
+                attri_manager.find_attribute_store( attribute_names[i] );
+
+            if( dynamic_cast< ContinuousProperty* >( properties[i] )
+                != nullptr )
+            {
+                ContinuousProperty* continuous =
+                    static_cast< ContinuousProperty* >( properties[i] );
+
+                continuous->pushBackDoubleHdf5Array1dOfValues(
+                    (double*) store->data(), store->size() * store->dimension(),
+                    hdf_proxy_ );
+            }
+            else if( dynamic_cast< DiscreteProperty* >( properties[i] )
+                     != nullptr )
+            {
+                DiscreteProperty* discrete =
+                    static_cast< DiscreteProperty* >( properties[i] );
+
+                discrete->pushBackIntHdf5Array1dOfValues( (int*) store->data(),
+                    store->size() * store->dimension(), hdf_proxy_, -99999 );
+            }
+            else
+            {
+                ringmesh_assert_not_reached;
+            }
+        }
+        return true;
+    }
+
+    bool GeoModelAdapterRESQMLImpl::write_surfaces()
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this function to reduce its Cognitive Complexity from 27 to the 25 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS3776)